datadog_checks_dev/datadog_checks/dev/tooling/e2e/local.py

```diff
@@ -0,0 +1,128 @@
+)
+from .platform import LINUX, MAC, WINDOWS
+from ..constants import get_root
+from ...utils import ON_MACOS, ON_WINDOWS, ON_LINUX, path_join, file_exists
+from ...subprocess import run_command
+
+
+class LocalAgentInterface(object):
+    def __init__(self, check, env, base_package=None, config=None, metadata=None, agent_build=None, api_key=None):
+        self.check = check
+        self.env = env
+        self.base_package = base_package
+        self.config = config or {}
+        self.metadata = metadata or {}
+        self.agent_build = agent_build
+        self.api_key = api_key or FAKE_API_KEY
+
+        self._agent_version = self.metadata.get('agent_version')
+        self.config_dir = locate_config_dir(check, env)
+        self.config_file = locate_config_file(check, env)
+        self.config_file_name = config_file_name(self.check)
+
+    @property
+    def platform(self):
+        if ON_LINUX:
+            return LINUX
+        elif ON_MACOS:
+            return MAC
+        elif ON_WINDOWS:
+            return WINDOWS
+        else:
+            raise Exception("Unsupported OS for Local E2E")
+
+    @property
+    def agent_version(self):
+        return self._agent_version or DEFAULT_AGENT_VERSION
+
+    @property
+    def agent_command(self):
+        return '{}'.format(get_agent_exe(self.agent_version, platform=self.platform))
+
+    def write_config(self):
+        write_env_data(self.check, self.env, self.config, self.metadata)
+        self.copy_config_to_local_agent()
+
+    def remove_config(self):
+        remove_env_data(self.check, self.env)
+        self.remove_config_from_local_agent()
+
+    def copy_config_to_local_agent(self):
+        check_conf_file = os.path.join(
+            get_agent_conf_dir(self.check, self.agent_version, self.platform),
+            '{}.yaml'.format(self.check)
+        )
+        if file_exists(check_conf_file):
+            copyfile(check_conf_file, '{}.bak'.format(check_conf_file))
+        copyfile(self.config_file, check_conf_file)
+
+    def remove_config_from_local_agent(self):
+        check_conf_file = os.path.join(
+            get_agent_conf_dir(self.check, self.agent_version, self.platform),
+            '{}.yaml'.format(self.check)
+        )
+        backup_conf_file = '{}.bak'.format(check_conf_file)
+        os.remove(check_conf_file)
+        if file_exists(backup_conf_file):
+            move(backup_conf_file, check_conf_file)
+
+    def run_check(self, capture=False, rate=False):
+        command = '{} check {}{}'.format(
+            self.agent_command,
+            self.check,
+            ' {}'.format(get_rate_flag(self.agent_version)) if rate else ''
+        )
+        return run_command(command, capture=capture)
+
+    def update_check(self):
+        install_cmd = AGENT_CMD[self.platform]['pip'] + ['install', '-e',  path_join(get_root(), self.check)]
+        return run_command(install_cmd, capture=True, check=True, shell=True)
+
+    def update_base_package(self):
```

Comment:
With a local agent, I'm not sure there is a really clean way to accomplish this without reinstalling the agent from scratch. Thats a bit challenging to do in ddev because it seems the one line install script doesn't allow you to specify an agent version to grab. 

As discussed offline, I'll add a clear message that this won't be undone after e2e teardown.